### PR TITLE
Add forms for posts and comments, Add a button to update likes.

### PR DESCRIPTION
**Pull request by jchernandez87**
### Changes: 😄 

- Enable methods new and create for resources :posts.
- Add a form to create new posts.
- Add a route using resources and methods new and create for comments.
- Add comments controller and add new and create actions.
- Add comments new view.
- Add a form to create new comments.
- Add a route with the method put for likes as like.
- Update posts controller with an action for like,
- Add a button to _post partial to update likes when is clicked.

### CODE REVIEW

File: app/views/comments/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    <div>
+      <%= f.label :text, class: "font-mid mid-gap" %><br>
+      <%= f.text_area :text, size: "70x5", class: "border mid-gap", :required => true %>
+    </div>
+
+    <div>
+      <%= f.submit "Add comment", class: "btn" %>
+    </div>
+  <% end %>
+</div>
```
Comment (andersonlebon) [inline app/views/comments/new.html.erb:13]: - [ ] Kindly provide a button link to this form in your post#ishow view currently I am accessing this form by only modifying the route. 
example
```html.erb
        <%= link_to 'Create new comment', new_user_post_comment_path(@user.id), class: "btn btn-dark" %>
```

File: app/views/posts/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    
+    <div>
+      <%= f.label :text, class: "font-mid mid-gap" %><br>
+      <%= f.text_area :text, size: "70x5", class: "border mid-gap", :required => true %>
+    </div>
+
+    <div>
+      <%= f.submit "Add post", class: "btn" %>
+    </div>
+  <% end %>
```
Comment (andersonlebon) [inline app/views/posts/new.html.erb:17]: Also here
- [ ] Kindly provide a button link to this form in your post#ishow view currently I am accessing this form by only modifying the route. 
example
```html.erb
        <%= link_to 'Create new post', new_user_post_path(@user.id), class: "btn btn-dark" %>
```

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -7,7 +9,44 @@ def index
...
+      render :new, locals: { post: @post }
+    end
+  end
+
+  def like
+    user = User.find(params[:user_id])
+    post = user.posts.find(params[:id])
+    Like.create(author_id: user.id, post_id: post.id)
+    redirect_to user_post_path(user, post)
+  end
```
Comment (andersonlebon) [inline app/controllers/posts_controller.rb:41]: - [ ] Nice job :clap:  on implementing this feature however, Kindly note that a user can only have one like on a post. You can address this issue by adding a condition here like :point_down: 
example in the likes_controllers:
```rb
class LikesController < ApplicationController
  def create
    @current_user = User.find(params[:user_id])
    @current_post = Post.find(params[:post_id])
    is_liked = Like.where(user: @current_user, post: @current_post).exists?
    
    if is_liked
      redirect_to user_post_url({ id: @current_post.id })
      flash.now[:error] = 'Error: You already liked this post'
    else
      @like = Like.new(user: @current_user, post: @current_post)
      @like.save
      flash[:success] = 'You liked this post'
    end
  end
end

```

File: app/views/shared/_post.html.erb [hunk 4]
```diff
@@ -4,4 +4,5 @@
     <p><%= post.text %></p>
   </div>
   <span class="post-stats">Comments: <%= post.comments_counter %>, Likes: <%= post.likes_counter %></span>
+  <%= button_to 'Like', like_path(user_id: @user, id: post), method: "put" %>
 </article>
```
Comment (andersonlebon) [inline app/views/shared/_post.html.erb:8]: - [ ] Kindly note that each post is the like to the post#show page

example :

```html.erb

 <%= link_to  user_path(user), class: "d-flex m-4 justify-content-between align-items-center" do %>

           .......................content
<% end %>

````

**Review by andersonlebon — CHANGES_REQUESTED** [review]
### Hi @jchernandez87 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bimbolabuari — APPROVED** [review]
# Project Approved 🚀

Hi @jchernandez87 🥇,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**